pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions_test.go

```diff
@@ -478,6 +480,188 @@ func TestAccessModeConflicts(t *testing.T) {
 	}
 }
 
+func Test_isSchedulableAfterPodDeleted(t *testing.T) {
+	GCEDiskVolState := v1.Volume{
+		VolumeSource: v1.VolumeSource{
+			GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+				PDName: "foo",
+			},
+		},
+	}
+	GCEDiskVolState2 := v1.Volume{
+		VolumeSource: v1.VolumeSource{
+			GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+				PDName: "bar",
+			},
+		},
+	}
+
+	AWSDiskVolState := v1.Volume{
+		VolumeSource: v1.VolumeSource{
+			AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+				VolumeID: "foo",
+			},
+		},
+	}
+	AWSDiskVolState2 := v1.Volume{
+		VolumeSource: v1.VolumeSource{
+			AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+				VolumeID: "bar",
+			},
+		},
+	}
+
+	RBDDiskVolState := v1.Volume{
```

Comment:
The rbd volume was deprecated in v1.28, non-functional in v1.31+. 

- https://github.com/kubernetes/kubernetes/pull/124546 removed the in-tree rbd volume plugin and its csi migration.

Should we remove all the rbd related code from the codebase but only keep the definition in API types.go? 

/cc @liggitt @msau42 @jsafrane @humblec @xing-yang 

Can you take a look at this? Thanks.

Reply:
we should not remove code guarding the API surface for the RBD fields

Reply:
I wonder about the little bits of support code which reference now-removed volumes, like `VisitPodSecretNames()` or `VisitPVSecretNames()`.   Or this scheduler restriction.  The volume driver is gone, should we still be enforcing mutual-exclusion of pods using the same volume of this flavor?